### PR TITLE
Update Angular and TypeScript quickstarters - master

### LINKED ISSUE #897: TypeScript QS fails to build
**Describe the bug**
First build of the TypeScript Quickstarter fails to build.

**To Reproduce**
1. Provision a Typescript Quickstarter
2. See error in the 1º build on component pipeline

**Expected behavior**
Quickstarter should be able to successfully build the 1º time.

**Affected version:**
 - OpenShift: [4.11]
 - OpenDevStack [4.2]

**Log Output**
```
+ npm install
npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
npm WARN deprecated w3c-hr-time@1.0.2: Use your platform's native performance.now() and performance.timeOrigin.
npm WARN deprecated source-map-url@0.4.1: See https://github.com/lydell/source-map-url#deprecated
npm WARN deprecated source-map-resolve@0.5.3: See https://github.com/lydell/source-map-resolve#deprecated
npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
npm WARN deprecated sane@4.1.0: some dependency vulnerabilities fixed, support for node < 10 dropped, and newer ECMAScript syntax/features added
npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.

added 606 packages in 31s

29 packages are looking for funding
  run `npm fund` for details
[Pipeline] sh
+ npm run build

> type-three@1.0.0 build
> tsc

node_modules/@types/babel__traverse/index.d.ts(68,50): error TS1005: ']' expected.
node_modules/@types/babel__traverse/index.d.ts(68,53): error TS1005: ';' expected.
node_modules/@types/babel__traverse/index.d.ts(68,58): error TS1005: ';' expected.
node_modules/@types/babel__traverse/index.d.ts(68,70): error TS1011: An element access expression should take an argument.
node_modules/@types/babel__traverse/index.d.ts(68,83): error TS1005: ';' expected.
node_modules/@types/babel__traverse/index.d.ts(68,84): error TS1128: Declaration or statement expected.
node_modules/@types/babel__traverse/index.d.ts(68,88): error TS1128: Declaration or statement expected.
[Pipeline] }
[Pipeline] // stage
[Pipeline] }
[Pipeline] // wrap
[Pipeline] stage
[Pipeline] { (odsPipeline error)
[Pipeline] echo
WARN: [type-three] ***** Finished ODS Pipeline for type-three (with error) ***** (took 69184 ms)
```

**Additional context**
I was able to avoid this error updating the typescript dependency into the package.json to version "^4.9.3".
Should we just update it by default for all QS in the [Jenkinsfile](https://github.com/opendevstack/ods-quickstarters/blob/master/be-typescript-express/Jenkinsfile#L27) used for the provisioning?

@sino92 @cschweikert

### PR DESCRIPTION
**Pull request by braisvq1996**
Maintenance of Angular and TypeScript quickstarters.

Bump packages version and update base images.
Fix auth configuration in nodejs18 agent for npm9 and karma configuration generation change in angular qs

related with #899
Fix #897 

Tasks: 
- [x] Provision sucessfull with Nodejs18 agent
- [x] Build and Deploy successfully with Nodejs18 agent

### CODE REVIEW

File: be-typescript-express/Jenkinsfile [hunk 1]
```diff
@@ -24,17 +24,17 @@ odsQuickstarterPipeline(
...
+      def expressVersion = "4.18.2"
+      def jestVersion = "29.5.0"
+      def jestJunitVersion = "15.0.0"
 
       sh "npm i typescript@$typescriptVersion express@$expressVersion @types/node@$nodeTypeVersion jest@$jestVersion @types/jest@$jestVersion"
       sh "npm i jest-junit@$jestJunitVersion --save-dev"
 
       print "Generate ts config file. Skip library check and redirect transpiled ts files to dist folder."
-      print "Target 10.0.0 and newer versions. Tell the compiler its okay to output JS syntax features from ES2018."
+      print "Target 15.0.0 and newer versions. Tell the compiler its okay to output JS syntax features from ES2018."
```
Comment (cschweikert) [inline be-typescript-express/Jenkinsfile:37]:
```suggestion
      print "Target 15.0.0 and newer versions. Tell the compiler it is okay to output JS syntax features from ES2018."
```

File: fe-angular/Jenkinsfile [hunk 2]
```diff
@@ -14,7 +14,7 @@ node {
 
 library("ods-jenkins-shared-library@${sharedLibraryRef}")
 
-def angularCliVersion = "13.3.0"
+def angularCliVersion = "15.2.4"
```
Comment (cschweikert) [inline fe-angular/Jenkinsfile:17]: Latest version has changed a few hours ago (https://www.npmjs.com/package/@angular/cli). How about
```suggestion
def angularCliVersion = "15.2.7"
```

File: fe-angular/files/docker/Dockerfile [hunk 3]
```diff
@@ -1,4 +1,5 @@
-FROM nginx:1.21.6-alpine
+#FROM nginx:1.21.6-alpine
```
Comment (cschweikert) [inline fe-angular/files/docker/Dockerfile:1]: maybe remove the unused code here

File: be-typescript-express/Jenkinsfile (hunk 1 above)
Comment (braisvq1996) [reply be-typescript-express/Jenkinsfile]: Already changed

File: fe-angular/Jenkinsfile (hunk 2 above)
Comment (braisvq1996) [reply fe-angular/Jenkinsfile]: The test were successfull so version updated :)

**Review by cschweikert — APPROVED** [review]
🚀

**Pull request merged**